FaunaDB.Client/Types/RefV.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using FaunaDB.Query;
+using Newtonsoft.Json;
+
+namespace FaunaDB.Types
+{
+    /// <summary>
+    /// A FaunaDB ref type.
+    /// <para>
+    /// See <see href="https://fauna.com/documentation/queries#values-special_types">FaunaDB Special Types</see>.
+    /// </para>
+    /// </summary>
+    public class RefV : ScalarValue<string>
+    {
+        /// <summary>
+        /// Create a Ref from a string, such as <c>new Ref("databases/prydain")</c>.
+        /// </summary>
+        /// <param name="value">Value.</param>
+        public RefV(string value) : base(value)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+        }
+
+        protected internal override void WriteJson(JsonWriter writer) =>
+            writer.WriteObject("@ref", Value);
+    }
+
+    public abstract class ObjectRef : RefV
```

Comment:
I'm over the fence with this new abstractions. I would like to hear more about your motivation to create them.

He is my opinion:

As we create more classes to represent references, we expand the  required vocabulary the developer needs to know in order to use Fauna. It also give them the impression that classes, databases, keys are all different types of references instead of composed structured references. It increases complexity as now the driver must handle different types in order to express things that are actually the same thing.

Purity of abstraction would by us some simplicity, IMO. In the other hand, we also need some syntax sugar.

I was expecting something more on the lines of:

```csharp
new RefV(
  class: new RefV(BuiltInClasses.CLASSES, "my-class"),
  id: "123456789"
)
```

Perhaps a little more sugar on top of that constructor:

```charp
RefV.Of(
  class: RefV.Class("my-class"),
  id: "123456789"
)
```

Please, let me know your thoughts.